index.bs

```diff
@@ -378,7 +378,7 @@ parallel</a>:
 1. Run the <a>exit Picture-in-Picture algorithm</a>.
 2. If the previous step threw an exception, reject |promise| with that
     exception and abort these steps.
```

Comment:
```suggestion
2. If the previous step [=exception/throws|threw=] an exception, [=queue a task=] using [=this=]'s [=media element event task source=] to [=/reject=] |promise| with that exception and abort these steps.
```

Reply:
Both WebKit and Chrome do not queue a task from what I can see at https://github.com/WebKit/WebKit/blob/d36ed108e92fb350723da8e7a88e7a7e8be1ef45/Source/WebCore/Modules/pictureinpicture/DocumentPictureInPicture.cpp#L54 and https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/picture_in_picture/document_video_picture_in_picture.cc;l=43;drc=95cf77bcd2a1f44e564e8404d2d363f508d580ce

I'm not sure we want to update both implementations for this.

Reply:
Wait! Then this algorithm is wrong then. Above it says it runs “in parallel”, and you can see from both implementations that it’s running on the main thread. 

This needs be updated to return a promise rejected with X and the later run the steps in parallel.

Reply:
The reject path runs on the main thread because it's easy to catch some errors but the resolve path and some reject paths are not.

Reply:
We need to split those up then. This should match implementations. I can see like 6 reject cases that run on the main thread. Only see this in WebKit getting sent off main thread: 

```C++ 
    if (videoElement.webkitSupportsPresentationMode(HTMLVideoElement::VideoPresentationMode::PictureInPicture)) {
        videoElementPictureInPicture->m_enterPictureInPicturePromise = WTFMove(promise);
        videoElement.webkitSetPresentationMode(HTMLVideoElement::VideoPresentationMode::PictureInPicture);
    }
```

Reply:
Even there... we would need to check what videoElement.webkitSetPresentationMode does and if it goes off main thread.